native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -242,7 +243,11 @@ private static BMap<BString, Object> insertDataToMap(BMap<BString, Object> child
             if (children instanceof BMap) {
                 put(rootNode, keyValue, children);
             } else if (children == null) {
-                putAsFieldTypes(rootNode, keyValue, EMPTY_STRING, fieldType);
+                if (fieldType instanceof ReferenceType && TypeUtils.getReferredType(fieldType) instanceof RecordType) {
+                    put(rootNode, keyValue, ValueCreator.createMapValue(Constants.JSON_MAP_TYPE));
+                } else {
+                    putAsFieldTypes(rootNode, keyValue, EMPTY_STRING, fieldType);
```

Comment:
what happens if we pass null, instead of empty string here

Reply:
According to this [doc](https://www.w3schools.com/xml/xml_elements.asp#:~:text=Empty%20XML%20Elements,is%20said%20to%20be%20empty.&text=The%20two%20forms%20produce%20identical,Empty%20elements%20can%20have%20attributes.), An element with no content is said to be empty.